cw_core/lib/erc20_token.dart

```diff
@@ -74,9 +74,7 @@ class Erc20Token extends CryptoCurrency with HiveObjectMixin {
   static const arbitrumBoxName = 'ArbitrumErc20Tokens';
 
   @override
-  bool operator ==(other) =>
-      (other is Erc20Token && other.contractAddress == contractAddress) ||
-      (other is CryptoCurrency && other.title == title);
+  bool operator ==(Object other) => other is Erc20Token && other.contractAddress == contractAddress;
```

Comment:
why is this change?

Reply:
It affected the indexOf operator, which makes use of the "==" operator for comparison.

Based on the previous override, which adds `|| (other is CryptoCurrency && other.title == title);`, it means that any cryptocurrency that had the smae title compared the same, regardless of the chain/tag.

That's what caused one of the bugs Jack reported earlier, that USDC(Base) was returned, even though he selected USDC(Arb).

So indexOf stops at the first USDC(which was base), because it satisfies its check.